tests/lean/run/reflected.lean

```diff
@@ -1,4 +1,15 @@
-meta def eval_list (α : Type) [reflected α] (e : expr) : tactic (list α) :=
+meta def eval_list (α : Type) [reflected _ α] (e : expr) : tactic (list α) :=
 tactic.eval_expr (list α) e
 
 run_cmd eval_list ℕ ↑`([1, 2]) >>= tactic.trace
+
+def synonym (α : Type) : Type := α
+
+def synonym.of {α : Type} (a : α) : synonym α := a
+
+meta instance synonym.reflect {α : Type} [h : has_reflect α] [reflected _ α] :
+  has_reflect (synonym α) :=
+λ x, show reflected _ (synonym.of x), from `(synonym.of).subst (h x)
+
+run_cmd let x := synonym.of 1 in tactic.trace `(x)
```

Comment:
> Did you actually run into any issues with the wrong instance or was it just cosmetics? From what I can tell, even if you get the wrong instance the expr should still be defeq to the expected one.

The expr is defeq, but I assume it won't pick up the right typeclass instances if you antiquote it into an expression because its inferred type is wrong.

Reply:
Moved, good call